src/listener.ts

```diff
@@ -0,0 +1,24 @@
+import { NestFactory } from '@nestjs/core';
+import { MicroserviceOptions, Transport } from '@nestjs/microservices';
+import { AppModule } from './app.module';
+
+/**
+ * It creates a microservice that listens to a RabbitMQ queue and uses the AppModule to handle incoming messages
+ */
+async function bootstrap() {
+  const app = await NestFactory.createMicroservice<MicroserviceOptions>(
+    AppModule,
+    {
+      transport: Transport.RMQ,
+      options: {
+        urls: [process.env.RABBITMQ_URL],
+        queue: process.env.RABBITMQ_QUEUE,
+        queueOptions: {
+          durable: false,
```

Comment:
i thought durable is meant to be set to true, in case the rabbitMQ service itself goes down so it still keeps records of queues unattended to or unprocessed.
Either ways it should still work fine

Reply:
It was set to true initially, but for some reasons messages wasn't delivered

Reply:
> It was set to true initially, but for some reasons messages wasn't delivered, the service won't even start

Fixed now, I found out that if a queue's durability is initially set to false, you can't change it to true unless you change the queue name

Reply:
nice work bro